qa/SidechainTestFramework/sc_test_framework.py

```diff
@@ -53,39 +55,71 @@ def join_network(self):
+                    - address
+                    - connectionTimeout
+                    - reconnectionDelay
+                    - reconnectionMaxAttempts
+                    
+    Example: 2 mainchain nodes and 3 sidechain nodes (with default websocket configuration) bootstrapped, respectively, from mainchain node first, first, and third.
+    {
+        mc_node_1: {
+            1: [3, 1000, [ws://localhost:8888, 100, 1, 1]]
+            2: [3, 1000, [ws://localhost:8888, 100, 1, 1]]
+        }
+        mc_node_2: {
+            1: [3, 1000, [ws://localhost:8888, 100, 1, 1]]
+        }
+    }
+     
+     Output: a map of:
+     - key: i, i=[1,...,n] with n =t he number of sidechain nodes to be bootstrapped
+     - value: bootstrap information if the sidechain node i. An array of:
+        - sidechain_id
+        - account_secrets
+        - total_balance
+        - genesis_info
     
     NB: for each account add an amount of 100
     """
-    def bootstrap_sidechain(self, number_of_sidechains_nodes, number_of_accounts_per_sidechain, mainchain_node):
-        sc_nodes_bootstrap_info = {}
-        for i in range(number_of_sidechains_nodes):
-            n_keys = number_of_accounts_per_sidechain[i]
-            account_secrets = generate_secrets(i, n_keys)
-            balances = []
-            total_balance = 100*n_keys
-            for j in range(n_keys):
-                balances.append(100)
-            sidechain_id = "000000000000000000000000000000000000000000000000000000000000000{0}".format(i)
-            genesis_info = get_genesis_info(sidechain_id, mainchain_node, 1000, account_secrets, balances)
-            print "Sidechain created with id: " + sidechain_id
-            initialize_sc_datadir(self.options.tmpdir, i, account_secrets, genesis_info[0])
-            sc_nodes_bootstrap_info[i] = [sidechain_id, account_secrets, total_balance, genesis_info[1]]
-        return sc_nodes_bootstrap_info
+    def bootstrap_sidechain(self, network={}):
+        self.sc_nodes_bootstrap_info = {}
+        total_number_of_sidechains = 0
+        for mc_node in network:
+            mc_node_values = network[mc_node]
+            number_of_sidechains_nodes = len(mc_node_values)
+            for i in range(number_of_sidechains_nodes):
+                sc_node_info = mc_node_values[i]
+                sc_index = i + total_number_of_sidechains
+                number_of_accounts = sc_node_info[0]
+                withdrawal_epoch_length = sc_node_info[1]
+                websocket_conf = sc_node_info[2]
+                account_secrets = generate_secrets(sc_index, number_of_accounts)
+                balances = []
+                total_balance = 100*number_of_accounts
+                for j in range(number_of_accounts):
+                    balances.append(100)
+                sidechain_id = "000000000000000000000000000000000000000000000000000000000000000{0}".format(sc_index)
```

Comment:
do we need to calculate a different sc id? what if sc_index is greater than `F`?

Reply:
sc id is a convenient value used for distinguish the nodes.